pkg/services/ngalert/accesscontrol.go

```diff
@@ -0,0 +1,180 @@
+package ngalert
+
+import (
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/datasources"
+)
+
+var (
+	rulesReaderRole = accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Name:        accesscontrol.FixedRolePrefix + "alerting.rules:reader",
+			DisplayName: "Rules Reader",
+			Description: "Can read alert rules in all Grafana folders and external providers",
+			Group:       "Alerting",
+			Version:     1,
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: accesscontrol.ActionAlertingRuleRead,
+					Scope:  dashboards.ScopeFoldersAll,
+				},
+				{
+					Action: accesscontrol.ActionAlertingRuleExternalRead,
+					Scope:  datasources.ScopeDatasourcesAll,
+				},
+			},
+		},
+	}
+
+	rulesEditorRole = accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Name:        accesscontrol.FixedRolePrefix + "alerting.rules:editor",
+			DisplayName: "Rules Editor",
+			Description: "Can add,update and delete rules in any Grafana folder and external providers",
+			Group:       "Alerting",
+			Version:     1,
+			Permissions: accesscontrol.ConcatPermissions(rulesReaderRole.Role.Permissions, []accesscontrol.Permission{
```

Comment:
This ends up being every action under `alert.rules` and `alert.rules.external` right? Should the rule editor just get access to `alert.rules:*` and `alert.rules.external:*` so that it auto-inherits any new permissions we may add? This seems more in line with the spirit of what this role is supposed to do. (same throughout)

Reply:
I do not think wildcard actions are supported.

Reply:
in pkg/services/accesscontrol/models.go there are several wildcard actions defined above the ones we added - `dashboards:*`, `teams:*`, etc. We don't have to use them necessarily, though.